app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -69,13 +69,11 @@ export default class FileAttachmentList extends PureComponent {
     }
 
     componentDidUpdate(prevProps) {
-        if (prevProps.files !== this.props.files) {
-            this.filesForGallery = this.getFilesForGallery(this.props);
-            this.buildGalleryFiles().then((results) => {
-                this.galleryFiles = results;
-            });
-        }
-        if (this.props.files !== prevProps.files && this.props.files.length === 0) {
+        this.filesForGallery = this.getFilesForGallery(this.props);
+        this.buildGalleryFiles().then((results) => {
+            this.galleryFiles = results;
+        });
+        if (prevProps.files.length !== this.props.files.length) {
```

Comment:
In this case, we're comparing the length. In case it changes we ask for the fileinfo to the server, this way we are able to refresh the state but just once

Reply:
The actual culprit is the call to get the files in line 77, this check over here helps to avoid making further requests.

You could move  `this.buildGallleryFiles` and `this.filesForGallery = this.getFilesForGallery(this.props);` inside this code block as it should only apply when the files are different.

In truth, this is only a partial fix as the server is currently returning an empty array for files Infos and and empty array for files in the post **metadata** when it should not.

For the fileInfos it should return a 404 and for post metadata it should not include the `files` prop.

Reply:
@enahum Thank you for the review :tada: 

How about tackling this in two phases:

1. I'll move those lines inside the condition and try to fix this part as soon as possible (with this partial fix)
2. Create a new discussion and/or ticket with server people about the 404 and files prop removal because it is a sensitive matter and I do not completely agree about returning a 404 in our API when we're dealing with arrays.

Sounds good?

Reply:
Yes, for 1 is this PR, for two I’m ok with having a different conversation ticket for number 2